### PR TITLE
Make predictions on multiple image units

### LINKED ISSUE #770: bulk predict
predict should allow inferring catalogs for a large set of images (from one survey)

### PR DESCRIPTION
**Pull request by zhixiangteoh**
Allows making predictions on multiple `image_id`s (e.g., RCF, brick)

WIP:
- Tests (`test_sdss_reconstruct.py`)
- Predictions plots per image unit

Will fix #770, #859.

### CODE REVIEW

File: bliss/api.py [hunk 1]
```diff
@@ -133,14 +133,14 @@ def train_on_cached_data(
...
-        SDSSDownloader(run, camcol, field, self.cwd + f"/{download_dir}").download_all()
+        SDSSDownloader([(run, camcol, field)], self.cwd + f"/{download_dir}").download_all()
         # assert files downloaded at download_dir
 
     def predict_sdss(
         self,
         weight_save_path: str,
         **kwargs,
-    ) -> Tuple[FullCatalog, Table, Table]:
+    ) -> Tuple[FullCatalog, Table, Dict[Any, Table]]:
```
Comment (jeff-regier) [inline bliss/api.py:143]: Is `image_id` always an integer? If so, can we get rid of `Any` here?

**Review by jeff-regier — COMMENTED** [review]
Looks good so far!

File: bliss/api.py (hunk 1 above)
Comment (zhixiangteoh) [reply bliss/api.py:143]: Not always an integer. For example, for SDSS it is `Tuple[int, int, int]` for `(run, camcol, field)`; for DECaLS it is `str` for `brickname`.

**Comment by zhixiangteoh** [comment]
@XinyueLi1012 ptal and lmk if anything seems off, especially regarding the changes to predictions that touch code you've written

File: bliss/api.py [hunk 2]
```diff
@@ -149,19 +149,52 @@ def predict_sdss(
...
+        pred_tables = {}  # indexed by image_id
+        for image_id, pred in pred_for_image_id.items():
+            pred_tables[image_id] = pred_to_astropy_table(pred)
+        return est_cat, est_cat_table, pred_tables
+
+    def predict_decals(
+        self,
+        weight_save_path: str,
+        **kwargs,
+    ) -> Tuple[FullCatalog, Table, Dict[Any, Table]]:
```
Comment (jeff-regier) [inline bliss/api.py:172]: In api.py, can we merge `predict_sdss` and `predict_decals` into a single `predict` function? They look pretty similar.

Comment (jeff-regier) [reply bliss/api.py:172]: I suppose it's kind of beyond the scope of this PR though. Sometime to work on as part of #871 ? It'd be good to merge all this before we get merge conflicts.

**Review by jeff-regier — APPROVED** [review]

**Pull request merged**